File: 2020/src/AoC2020/AoCHelper.cs
using System.Collections.Generic;
using System.IO;

namespace AoC2020
{
    public static class AoCHelper
    {
        public static List<string> GetPuzzleInput(int day)
        {
            var paths = new string[] {"puzzle_inputs", $"day{day}_example.txt"};
            // var paths = new string[] {"puzzle_inputs", $"day{day}_example2.txt"};
            // var paths = new string[] {"puzzle_inputs", $"day{day}.txt"};
            var fullPath = Path.Combine(paths);
            var result = new List<string>();
            
            using (var reader = File.OpenText(fullPath))
            {
                var line = reader.ReadLine();

                while (line != null)
                {
                    result.Add(line);
                    line = reader.ReadLine();
                }
            }

            return result;
        }
    } 
}

File: 2021/src/com/adventofcode/AoCHelper.java
package com.adventofcode;

import java.io.File;
import java.io.FileNotFoundException;
import java.nio.file.Paths;
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class AoCHelper {
    public static List<String> getPuzzleInput(int day) {
//        var fullPath = Paths.get("puzzleInputs", String.format("day%s_example.txt", day));
        var fullPath = Paths.get("puzzleInputs", String.format("day%s.txt", day));
        var result = new ArrayList<String>();

        try {
            Scanner scanner = new Scanner(new File(fullPath.toString()));
            while (scanner.hasNextLine()) {
                result.add(scanner.nextLine());
            }
            scanner.close();

            return result;
        } catch (FileNotFoundException e) {
            e.printStackTrace();

            return null;
        }
    }
}
